ubuntudesign/documentation_builder/extensions.py

```diff
@@ -0,0 +1,157 @@
+            <span class="p-notification__status">Note:</span>
+            <span class="p-notification__line">Things are reasonable</span>
+          </p>
+        </div>
+
+        <div class="p-notification--warning">
+          <p class="p-notification__response">
+            <span class="p-notification__line">I have no title</span>
+          </p>
+        </div>
+
+        <div class="p-notification--positive">
+          <p class="p-notification__response">
+            <span class="p-notification__status">Awesome:</span>
+            <span class="p-notification__line">
+                Everything is now wonderful
+            </span>
+          </p>
+        </div>
+
+    This builds on Python Markdown's existing [admonition extension][2].
+
+    [1]: https://docs.vanillaframework.io/en/patterns/notification
+    [2]: https://pythonhosted.org/Markdown/extensions/admonition.html
+    """
+
+    def extendMarkdown(self, md, md_globals):
+        """ Add Notifications to Markdown instance. """
+        md.registerExtension(self)
+
+        md.parser.blockprocessors.add(
+            'notifications',
+            NotificationsProcessor(md.parser),
+            '_begin'
+        )
+
+
+class NotificationsProcessor(BlockProcessor):
+
+    type_classes = {
+        'note': 'p-notification',
+        'warning': 'p-notification--warning',
+        'positive': 'p-notification--positive',
+        'negative': 'p-notification--negative',
+    }
+    line_match = re.compile(r'(?:^|\n)!!!\ ?([\w\-]+)(?:\ "(.*?)")?')
+
+    def test(self, parent, block):
+        sibling = self.lastChild(parent)
+
+        return (
+            self.line_match.search(block) or
+            (
+                block.startswith(' ' * self.tab_length) and
+                sibling is not None and
+                sibling.get('class', '').startswith('p-notification')
+            )
+        )
+
+    def run(self, parent, blocks):
+        sibling = self.lastChild(parent)
+        block = blocks.pop(0)
+        match = self.line_match.search(block)
+
+        if match:
+            block = block[match.end() + 1:]  # removes the first line
+
+        block, theRest = self.detab(block)
+        contents = block.replace('\n', ' ').replace('\r', '').strip()
+
+        if match:
+            template = jinja2.Template(
+                '<div class="{{ class }}">' +
+                '  <p class="p-notification__response">' +
+                '    {% if title %}' +
+                '      <span class="p-notification__status">' +
+                '        {{ title }}:' +
+                '      </span>' +
+                '    {% endif %}' +
+                '    <span class="p-notification__line">{{body}}</span>' +
+                '  </p>' +
+                '</div>'
+            )
+            notification_type, title = self.get_type_and_title(match)
+
+            markup = template.render(
+                {
+                    'class': self.type_classes[notification_type],
```

Comment:
My only comment here is do we want to have a fallback default class (note?) - at the moment any non-matching type throws an error.